lib/logger/lib/logger/formatter.ex

```diff
@@ -62,20 +62,28 @@ defmodule Logger.Formatter do
   defp prune_binary(<<_, t::binary>>, acc), do: prune_binary(t, <<acc::binary, @replacement>>)
   defp prune_binary(<<>>, acc), do: acc
 
-  @doc ~S"""
-  Compiles a format string into a data structure that the `format/5` can handle.
+  @doc """
+  Compiles a format string into a data structure that `format/5` can handle.
+
+  Check the module doc for documentation on the valid parameters that
+  will be interpolated in the pattern. If you pass `nil` as the pattern,
+  the pattern defaults to:
+
+      #{inspect(@default_pattern)}
+
+  If you want to customize formatting through a custom formatter, you can
+  pass a `{module, function}` tuple as the `pattern`.
 
-  Check the module doc for documentation on the valid parameters. If you
-  pass `nil`, it defaults to: `$time $metadata [$level] $levelpad$message\n`
+      iex> Logger.Formatter.compile("$time $metadata [$level] $message\\n")
+      [:time, " ", :metadata, " [", :level, "] ", :message, "\\n"]
 
-  If you would like to make your own custom formatter simply pass
-  `{module, function}` to `compile/1` and the rest is handled.
+      iex> Logger.Formatter.compile({MyLoggerFormatter, :format})
+      {MyLoggerFormatter, :format}
 
-      iex> Logger.Formatter.compile("$time $metadata [$level] $message\n")
-      [:time, " ", :metadata, " [", :level, "] ", :message, "\n"]
   """
   @spec compile(binary | nil) :: [pattern | binary]
-  @spec compile({atom, atom}) :: {atom, atom}
+  @spec compile(pattern) :: pattern when pattern: {module, function :: atom}
```

Comment:
This *should* make dialyzer smarter here, right? :)

Reply:
Yes.